### DISCUSSION TITLE
/access-a-synology-nas-with-traefik-on-dsm7/

### DISCUSSION THREAD

**Discussion by migratecomments** [ironicbadger/blog.ktz.me#192] [Announcements]
Access a Synology NAS with Traefik on DSM7
        
[https://blog.ktz.me/access-a-synology-nas-with-traefik-on-dsm7/](https://blog.ktz.me/access-a-synology-nas-with-traefik-on-dsm7/)

**Comment by jhollowe** [comment]
Why do you have both a config file and the configuration in the `command` of the container? Is there a reason to split the config or have in one place or the other?

**Comment by CelticWebSolutions** [comment]
I've been trying to find a way of getting this up and running on my Synology NAS for some time.  I found your page and hoped it wood have something to enable me to get it sorted.  Unfortunately I keep getting Error starting userland proxy: listen tcp4 0.0.0.0:443: bind: address already in use. To my knowledge there isn't anything running on there that uses that address but I still don't seem to be able to bind to etc address.  

Are you aware of any way I can fix the please?

I look forward to your reply. 

Regards
CelticWebs

**Reply by agross** [reply]
`sudo ss -tlpn` shows which processes bind a port. If `ss` does not exist, you can try `netstat` with the same arguments.

**Reply by Leon7810** [reply]
DSM is already using port 443.

**Reply by agross** [reply]
You can use docker MACVLAN to run Traefik on port 443 on another IP address than the Synology.

**Reply by Leon7810** [reply]
Would you still be able to route other traffic through Traefik on a MacVLAN?

**Reply by teeesss** [reply]
This will fix your issue and you can replace port 82 and 444 with the ports you want to move default port 80 and 443 to.
Also set it up in Task Scheduler to run on startup.

`sed -i -e 's/80/82/' -e 's/443/444/' /usr/syno/share/nginx/server.mustache /usr/syno/share/nginx/DSM.mustache /usr/syno/share/nginx/WWWService.mustache`

`synosystemctl restart nginx`
